# Copyright (c) 2024 Advanced Micro Devices, Inc. All Rights Reserved.
syntax = "proto3";

package Schola;

message BoxSpace {

    message BoxSpaceDimension {
        float low = 1;
        float high = 2;
    }
    repeated BoxSpaceDimension dimensions = 1;
}

message DiscreteSpace {
    repeated int32 high = 1;
}

message BinarySpace {
    int32 shape = 1;
}

message FundamentalSpace {
    oneof space {
        BoxSpace box_space = 1;
        DiscreteSpace discrete_space = 2;
        BinarySpace binary_space = 3;
    }
}

message DictSpace {
    //We just send it as an array for ordering purposes
    //This will be converted to a dict on the python side
    repeated FundamentalSpace values = 1;
    repeated string labels = 2;
}
